src/ansys/fluent/core/solver/flobject.py

```diff
@@ -802,60 +801,55 @@ def get_cls(name, info, parent=None):
         else:
             cls = type(pname, (base,), dct)
 
-        children = info.get("children")
         taboo = set(dir(cls))
         taboo |= set(
             ["child_names", "command_names", "argument_names", "child_object_type"]
         )
-        if children:
-            taboo.add("child_names")
-            cls.child_names = []
-            for cname, cinfo in children.items():
+
+        doc = ""
+
+        def _process_cls_names(info_dict, names, write_doc=False):
+            nonlocal taboo
+            nonlocal cls
+
+            for cname, cinfo in info_dict.items():
                 ccls = get_cls(cname, cinfo, cls)
-                i = 0
                 ccls_name = ccls.__name__
+
+                i = 0
+                if write_doc:
+                    nonlocal doc
+                    th = ccls._state_type
+                    th = th.__name__ if hasattr(th, "__name__") else str(th)
+                    doc += f"    {ccls.__name__} : {th}\n"
+                    doc += f"        {ccls.__doc__}\n"
+
                 while ccls_name in taboo:
                     if i > 0:
                         ccls_name = ccls_name[: ccls_name.rfind("_")]
                     i += 1
                     ccls_name += f"_{str(i)}"
                 ccls.__name__ = ccls_name
                 # pylint: disable=no-member
-                cls.child_names.append(ccls.__name__)
+                names.append(ccls.__name__)
                 taboo.add(ccls_name)
                 setattr(cls, ccls.__name__, ccls)
+
+        children = info.get("children")
+        if children:
+            taboo.add("child_names")
+            cls.child_names = []
+            _process_cls_names(children, cls.child_names)
+
         commands = info.get("commands")
         if commands:
             cls.command_names = []
```

Comment:
I think it's a bit more Pythonic these days to write `cls.command_names = list()`

Reply:
Hi Dan, 
I will follow these instructions now onwards.
Thank you.